### ISSUE TITLE
Using @auth private option,  graphql response was "Unauthorized"

### ISSUE THREAD

**Issue by dyson-yamashita** [aws-amplify/amplify-js#4563]
** Which Category is your question related to? **
Amplify API
** What AWS Services are you utilizing? **
Cognito and AWS AppSync
** Provide additional details e.g. code snippets **
@aws-amplify/api: 1.1.3
@aws-amplify/auth: 1.3.3

I want to set below permissions to User model.
- owner: read, write, update, delete
- others(cognite user): read only

I wrote below schema.
_schema.graphql_
```
type User
  @model
  @auth(rules: [
    { allow: owner }
    { allow: private, operations: [read] }
])
{
  id: String!
  name: String!
  description: String
  owner: String!
}
```
"Unauthorized" error was retruned using getUser of graphqlOperation by others(cognito user) .
```
let getsUserResult = await API.graphql(graphqlOperation(getUser, {id: userId}))
let user = getUserResult.data.getUser
```

Resonse message:
- errorType: Unauthorized
- message: Not Authorized to access getUser on type Query

Please inform me of solutions when using private option?

**Comment by tkntobfrk** [comment]
I've also found this to be the case. I didn't dig into the problem at length.

My workaround was to use groups instead of private when doing multiauth. This seems to work fine.

````
type User
  @model
  @auth(rules: [
    { allow: owner },
    { allow: groups, groups: ["users"], operations: [read] }
])
{
  id: String!
  name: String!
  description: String
  owner: String!
}
````

**Comment by dyson-yamashita** [comment]
@tkntobfrk Thank you for your comment. It was work the workaround. 
If I will resolve the problem, I write here.

**Comment by dyson-yamashita** [comment]
I resolved this problem. 

Hidden operations are allowed. 
https://aws-amplify.github.io/docs/cli-toolchain/graphql?sdk=js#owner-authorization
If set `operations: [create, update, delete]`, _read_ is allowd in below sample.
When it uses `allow: groups`, it is nesessary same operations of "owner".

In fact I used `groups`, so hidden operation was disabled.
I confimed that others  could read and not update _User_ data.

_schema.graphql_
```
type User
  @model
  @auth(rules: [
    { allow: owner, operations: [create, update, delete] }
    { allow: groups, groups: ["Admin"], operations: [create, update, delete]  }
])
{
  id: String!
  name: String!
  description: String
  owner: String!
}
```
